src/review_local.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import os
+import re
+import sys
+from dataclasses import dataclass
+from typing import Collection, Iterable, Literal, Optional
+
+from injector import inject
+from tqdm import tqdm
+
+from config import Config, Rule
+from local_run_state import LocalReviewRunState
+from suggestions import Suggester
+from voting import map_vote, map_vote_to_log_level
+
+
+@dataclass
+class FileInfo:
+	path: str
+	contents: Optional[str] = None
+
+
+@dataclass
+class RunOptions:
+	paths: Collection[str]
+	severity: Literal['none'] | Literal['wait'] | Literal['REJECT']
+	fix: bool
+
+
+@inject
+@dataclass
+class LocalReviewer:
+	config: Config
+	logger: logging.Logger
+	suggester: Suggester
+
+	def get_files(self, paths: Iterable[str]) -> Collection[str]:
+		result = []
+		for path in paths:
+			if os.path.isdir(path):
+				result.extend(self.get_files(os.path.join(path, f) for f in os.listdir(path)))
+			else:
+				result.append(path)
+		return result
+
+	def review_file(self, state: LocalReviewRunState, path: str):
+		rules = self.config.get('rules')
+		assert rules is not None
+		f = FileInfo(path)
+		for rule in rules:
+			self.run_rule_for_file(state, f, rule)
+
+	def run(self, options: RunOptions) -> None:
+		if options.fix:
+			raise NotImplementedError("Modifying files is not supported yet.")
+		sev = map_vote(options.severity)
+		assert sev is not None, f"Expected severity to be a string for a vote. Got: \"{options.severity}\"."
+		paths = self.get_files(options.paths)
+		state = LocalReviewRunState()
+		for path in tqdm(paths,
+			desc="Reviewing files",
+			unit_scale=True, mininterval=2, unit=" files"
+		):
+			self.review_file(state, path)
+		if state.error_level <= sev:
+			self.logger.error("Exiting with error level 1 because problems were found.")
+			sys.exit(1)
+
+	def run_rule_for_file(self, state: LocalReviewRunState, file_info: FileInfo, rule: Rule) -> None:
```

Comment:
this is only for the whole file, we also need to handling content from 'git diff', right?